CouchDB tests broken

## Problem

https://travis-ci.org/FRosner/cluster-broccoli/jobs/298663440 was green but now it is red after rerunning it.

## CouchDB Logs

```
[notice] 2017-11-08T08:39:08.921307Z nonode@nohost <0.1911.0> 1167f0b4f6 localhost:5984 127.0.0.1 undefined GET /broccoli_instances 404 ok 50
[error] 2017-11-08T08:39:08.938321Z nonode@nohost <0.1911.0> acfe5db504 Request to create N=3 DB but only 1 node(s)
[notice] 2017-11-08T08:39:09.122826Z nonode@nohost <0.1911.0> acfe5db504 localhost:5984 127.0.0.1 undefined PUT /broccoli_instances 201 ok 196
[notice] 2017-11-08T08:39:09.375437Z nonode@nohost <0.1911.0> 3fa1a6e44c localhost:5984 127.0.0.1 undefined PUT /_config/broccoli/broccoli_instances 404 ok 1
[notice] 2017-11-08T08:39:09.522160Z nonode@nohost <0.1944.0> 88f590d7c2 localhost:5984 127.0.0.1 undefined GET /broccoli_instances 200 ok 39
[notice] 2017-11-08T08:39:09.523520Z nonode@nohost <0.1944.0> 985954b120 localhost:5984 127.0.0.1 undefined PUT /_config/broccoli/broccoli_instances 404 ok 1
[notice] 2017-11-08T08:39:09.535464Z nonode@nohost <0.1945.0> 48512d73aa localhost:5984 127.0.0.1 undefined GET /broccoli_instances 200 ok 2
[notice] 2017-11-08T08:39:09.547668Z nonode@nohost <0.1945.0> 7423b357d3 localhost:5984 127.0.0.1 undefined PUT /_config/broccoli/broccoli_instances 404 ok 1
[notice] 2017-11-08T08:39:09.774083Z nonode@nohost <0.1946.0> 8cd143442c localhost:5984 127.0.0.1 undefined GET /broccoli_instances 200 ok 1
[notice] 2017-11-08T08:39:09.778912Z nonode@nohost <0.1946.0> f27c7a0854 localhost:5984 127.0.0.1 undefined PUT /_config/broccoli/broccoli_instances 404 ok 1
```

## Broccoli Logs

```
Caused by: java.lang.IllegalArgumentException: Unable to aquire lock for CouchDB (http://localhost:5984/_config/broccoli/broccoli_instances)
	at de.frosner.broccoli.instances.storage.couchdb.CouchDBInstanceStorage.<init>(CouchDBInstanceStorage.scala:51)
	at de.frosner.broccoli.instances.storage.couchdb.CouchDBStorageModule.provideCouchDBInstanceStorage(CouchDBStorageModule.scala:34)
	at de.frosner.broccoli.instances.storage.couchdb.CouchDBStorageModule$$FastClassByGuice$$cee6be31.invoke(<generated>)
	at com.google.inject.internal.ProviderMethod$FastClassProviderMethod.doProvision(ProviderMethod.java:264)
	at com.google.inject.internal.ProviderMethod$Factory.provision(ProviderMethod.java:401)
	at com.google.inject.internal.ProviderMethod$Factory.get(ProviderMethod.java:376)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:194)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
	at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:205)
	at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:199)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1085)
	at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:199)
	at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:180)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:110)
	at com.google.inject.Guice.createInjector(Guice.java:99)
	at com.google.inject.Guice.createInjector(Guice.java:84)
	at play.api.inject.guice.GuiceBuilder.injector(GuiceInjectorBuilder.scala:181)
	at play.api.inject.guice.GuiceApplicationBuilder.build(GuiceApplicationBuilder.scala:123)
	at play.api.inject.guice.GuiceApplicationLoader.load(GuiceApplicationLoader.scala:21)
	at play.core.server.ProdServerStart$.start(ProdServerStart.scala:47)
	at play.core.server.ProdServerStart$.main(ProdServerStart.scala:22)
	at play.core.server.ProdServerStart.main(ProdServerStart.scala)
```

## Solution

Looks like Broccoli is incompatible with CouchDB 2.

- Fix CouchDB version to 1.7.0 in the tests
- Document that we [only run with 1.7.0](https://github.com/FRosner/cluster-broccoli/wiki/Instances#couchdb-storage-couchdb)

--- a/http-api-tests/instance-persistence-couchdb/before-each
+++ b/http-api-tests/instance-persistence-couchdb/before-each
@@ -1,6 +1,6 @@
 #!/bin/bash
-docker pull couchdb
-docker run --rm -d --net host couchdb > couchdb.did
+docker pull couchdb:1.7.0
+docker run --rm -d --net host couchdb:1.7.0 > couchdb.did
 
 sleep $BROCCOLI_SLEEP_MEDIUM
 

